webviz_ert/data_loader/__init__.py

```diff
@@ -79,10 +130,81 @@ def get_parameter_data_url(
     return f"{server_url}{parameter_url}"
 
 
-def get_schema(api_url: str, project_id: Optional[str] = None) -> Mapping[str, Any]:
-    logging.info(f"Getting json from {api_url}...")
-    http_response = _requests_get(api_url, auth=get_auth(project_id))
-    http_response.raise_for_status()
+from typing import MutableMapping, Tuple, Optional
+
+ServerIdentifier = Tuple[str, Optional[str]]  # (baseurl, optional token)
+
+
+class DataLoader:
+    _instances: MutableMapping[ServerIdentifier, "DataLoader"] = {}
+
+    baseurl: str
+    token: str
+    _graphql_cache: MutableMapping[str, MutableMapping[dict, Any]]
+
+    def __new__(cls, baseurl: str, token: Optional[str] = None) -> "DataLoader":
+        if (baseurl, token) in cls._instances:
```

Comment:
Not sure that I understand this branch here.

Reply:
I want exactly one `DataLoader` object for each `baseurl` and `token` pair. `DataLoader` should be able to do caching before this PR is merged.

Using `__new__` instead of `__init__` because it still allows us to use `DataLoader(..., ...)` to initialise the object, _but_ `__new__` is called before the object is instantiated, so we can do instantiation manually and return the same object if the arguments are the same. `__init__`, on the other hand, first allocated the objects before calling `__init__` to instantiate it.